ValveResourceFormat/IO/GltfModelExporter.cs

```diff
@@ -185,24 +342,116 @@ public void ExportToFile(string resourceName, string fileName, VMesh mesh)
         {
             DstDir = Path.GetDirectoryName(fileName);
 
-            var exportedModel = ModelRoot.CreateModel();
-            exportedModel.Asset.Generator = GENERATOR;
+            var exportedModel = CreateModelRoot(resourceName, out var scene);
             var name = Path.GetFileName(resourceName);
-            var scene = exportedModel.UseScene(name);
+            var node = AddMeshNode(exportedModel, scene, null, name, mesh, null);
+
+            if (node != null)
+            {
+                // Swap Rotate upright, scale inches to meters.
+                node.WorldMatrix = TRANSFORMSOURCETOGLTF;
+            }
+
+            WriteModelFile(exportedModel, fileName);
+        }
 
-            var exportedMesh = CreateGltfMesh(name, mesh, exportedModel, false);
-            var meshNode = scene.CreateNode(name)
-                .WithMesh(exportedMesh);
+        private Node AddMeshNode(ModelRoot exportedModel, Scene scene, VModel model, string name,
+            VMesh mesh, Skeleton skeleton, string skinMaterialPath = null)
+        {
+            if (mesh.GetData().GetArray("m_sceneObjects").Length == 0)
+            {
+                return null;
+            }
+
+            var hasJoints = skeleton != null && skeleton.AnimationTextureSize > 0;
+
+            if (MeshNameCounterDict.TryGetValue(name, out var counter))
+            {
+                // Give it a unique name regardless
+                var existingNode = scene.FindNode(n => n.Name == name);
+                name = $"{name}.{counter++:000}";
```

Comment:
Does this counter increment even update in `MeshNameCounterDict`?

Reply:
You're right, not sure how I missed that one.